packages/authentication/android/src/main/java/dev/robingenz/capacitorjs/plugins/firebase/authentication/handlers/GoogleAuthProviderHandler.java

```diff
@@ -50,7 +55,20 @@ public void handleOnActivityResult(PluginCall call, ActivityResult result) {
             GoogleSignInAccount account = task.getResult(ApiException.class);
             String idToken = account.getIdToken();
             AuthCredential credential = GoogleAuthProvider.getCredential(idToken, null);
-            pluginImplementation.handleSuccessfulSignIn(call, credential, idToken);
+            ExecutorService background = Executors.newSingleThreadExecutor();
```

Comment:
I do not have much experience with `newSingleThreadExecutor` but it seems like `new Thread(job).start();` should be the better way (see this [SO answer](https://stackoverflow.com/a/19000809/6731412)).

I had implemented this feature before but was not satisfied with my implementation at the time so I removed it: https://github.com/robingenz/capacitor-firebase/pull/55/commits/10baf2875274dc78ff5780dbe46e39b197d616ba

Please try to move the code to create the access token to a seperate method.

Reply:
Got it, will change.

Reply:
Okay, I switched it to use `Thread`, but I'm not sure about moving the code to a separate method. This separate method would only ever run right after `handleOnActivityResult`, so separating it into another method doesn't really provide any benefit. Additionally, a call to it would require `call`, `credential`, `idToken`, and `account` since it's async and thus can't return a value back to `handleOnActivityResult` (would block UI thread). I think, thus, it would be better for the code to stay where it is. Happy to change it if you think otherwise, though.

Reply:
Okay, convinced 😄